### DISCUSSION TITLE
How to change the FLASH size in the linker script?

### DISCUSSION THREAD

**Discussion by romainreignier** [ch32-rs/ch32-hal#19] [Q&A] [answered]
Hi,

I have a program for the ch32v003 but the linker step fail because it uses too much space on FLASH:

```
error: linking with `rust-lld` failed: exit status: 1
[...]
= note: rust-lld: error: section '.rodata' will not fit in region 'FLASH': overflowed by 2472 bytes
```

I would like to temporarily increase the FLASH section in the linker script in order to have my binary built, I will then be able to investigate what part is taking too much space with [cargo-bloat](https://github.com/RazrFalcon/cargo-bloat).
I know that such binary will not work on the real target.

I would like to provide a custom [`memory.x`](https://github.com/ch32-rs/ch32-metapac/blob/f7f6af2adb13b0f4eb8d714d70126140c2283210/src/chips/ch32v003f4u6/memory_x/memory.x) file with a bigger size. But how to override the build flags in `build.rs` in order to use my local version?

Because all the ch32v003 variants have the same memory, I cannot switch my feature flag to another variant. Changing the family changes the qingke model to another version and the implementation is different (delay for example).

**Comment by andelf** [comment, marked as answer]
Refer: https://github.com/ch32-rs/ch32-hal/blob/main/examples/ch32v203/build.rs

**Reply by romainreignier** [reply]
Great, thank you, it works!
